src/port1.0/porttrace.tcl

```diff
@@ -213,18 +213,29 @@ namespace eval porttrace {
         }
 
         # Allow access to some Xcode specifics
-        allow trace_sandbox "/var/db/xcode_select_link"
-        allow trace_sandbox "/var/db/mds"
-        allow trace_sandbox [file normalize ~${macportsuser}/Library/Preferences/com.apple.dt.Xcode.plist]
-        allow trace_sandbox "$env(HOME)/Library/Preferences/com.apple.dt.Xcode.plist"
+        set xcode_paths {}
+        lappend xcode_paths "/var/db/xcode_select_link"
```

Comment:
Not sure which ones should be hidden or not

Reply:
Unfortunately I don't know either. You can do the following to find out:

Modify your `configure.cmd clang -v` Portfile to explicitly set `configure.env DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer` (and leave `use_xcode` at the default of `no`). If you run this Portfile in trace mode, it should fail. If it doesn't, we're not filtering the right things yet, or the filtering doesn't work.

Reply:
Actually, we should probably also include the entire Xcode installation location in the sandbox. Nothing should be used from `/Applications/Xcode.app/Contents/Developer` with `use_xcode no`.

Reply:
On Portfile:
```tcl
configure.env       DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
configure.cmd       clang -v
```
`port configure -t` correctly errors out with:
```
:info:configure Executing:  cd "/opt/macports-test/var/macports/build/_Users_firelink_git_macports-test/emptyport/work/emptyport-0.2" && clang -v --prefix=/opt/macports-test 
:debug:configure system:  cd "/opt/macports-test/var/macports/build/_Users_firelink_git_macports-test/emptyport/work/emptyport-0.2" && clang -v --prefix=/opt/macports-test 
:info:configure xcrun: error: invalid DEVELOPER_DIR path (/Applications/Xcode.app/Contents/Developer), missing xcrun at: /Applications/Xcode.app/Contents/Developer/usr/bin/xcrun
:info:configure Command failed:  cd "/opt/macports-test/var/macports/build/_Users_firelink_git_macports-test/emptyport/work/emptyport-0.2" && clang -v --prefix=/opt/macports-test 
:info:configure Exit code: 1
:info:configure The following file would have been hidden from the build system by trace mode if it existed:
:info:configure   /Applications/Xcode.app/Contents/Developer/mach_kernel
:error:configure Failed to configure emptyport: configure failure: command execution failed
:debug:configure Error code: NONE
:debug:configure Backtrace: configure failure: command execution failed
:debug:configure     while executing
:debug:configure "$procedure $targetname"
:error:configure See /opt/macports-test/var/macports/logs/_Users_firelink_git_macports-test/emptyport/main.log for details.
```
So I think `/Applications/Xcode.app/Contents/Developer` is already hidden in the sandbox courtesy of this code:
```tcl
set ddsplit [file split [file normalize [file join ${developer_dir} ".." ".."]]]
if {[llength $ddsplit] > 2 && [lindex $ddsplit end-1] eq "Contents" && [lindex $ddsplit end] eq "Developer"} {
    set ddsplit [lrange $ddsplit 0 end-2]
}
lappend xcode_paths [file join {*}$ddsplit]
```
I also tested that trace mode doesn't hide CommandLineTools. On the same Portfile with only `configure.cmd       clang -v`, `port configure -t` runs successfully so it should be OK.